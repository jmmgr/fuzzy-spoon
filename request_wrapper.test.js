'use strict';

const axios = require('axios');
const request_wrapper = require('./request_wrapper');

describe('request_wrapper', () => {
	test('on 404, verify we call handle404', async () => {
		const axios_mock = jest.spyOn(axios, 'request').mockRejectedValue({
			response: {
				status: 404,
				message: 'invalid url',
			},
		});

		const spy_handle404 = jest.spyOn(request_wrapper, 'handle404');
		await request_wrapper.request({
			method: 'get',
			url: 'https://randomURL',
		});
		expect(axios_mock).toHaveBeenCalled();
		expect(spy_handle404).toHaveBeenCalled();
		expect(spy_handle404.mock.calls).toMatchSnapshot('handle404_calls');
		expect(spy_handle404.mock.results).toMatchSnapshot('handle404_results');
	});
});
