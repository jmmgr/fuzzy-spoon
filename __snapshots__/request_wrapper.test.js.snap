// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`request_wrapper on 404, verify we call handle404: handle404_calls 1`] = `
Array [
  Array [
    Object {
      "method": "get",
      "url": "https://randomURL",
    },
    Object {
      "response": Object {
        "message": "invalid url",
        "status": 404,
      },
    },
  ],
]
`;

exports[`request_wrapper on 404, verify we call handle404: handle404_results 1`] = `
Array [
  Object {
    "isThrow": false,
    "value": Object {
      "error": "Reponse 404",
      "message": "Error, handling 404 response of url https://randomURL",
    },
  },
]
`;
